api/handlers.go

```diff
@@ -747,6 +754,8 @@ func postContainersStart(c *context, w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	container.Config.HostConfig = hostConfig
```

Comment:
you are erasing the hostconfig given at create, i thik we need to merge them instead.